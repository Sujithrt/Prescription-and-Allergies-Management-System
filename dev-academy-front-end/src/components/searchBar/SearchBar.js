import React, { useState } from 'react';
import './../searchBar/searchBar.css';
import { retrievePatients } from './../../api/PnamsService';
import AsyncSelect from 'react-select/async';
import makeAnimated from 'react-select/animated';
import { useHistory } from 'react-router-dom';
import { customStyles } from '../../Constants';


const animatedComponent = makeAnimated();

/**
 * Renders Search Bar in the Navigation Bar to search for patients
 * 
 */
export default function SearchBar() {    

    const history = useHistory();

    const [selected, setSelected] = useState("");

    /**
     * Loads list of patients from database and displays it to the user to select and navigate to patient profile
     * @param searchText 
     * @returns 
     */
    const loadPatients = async (searchText) => (
        retrievePatients(searchText)
            .then((response) =>
                response.data.map(i => ({ label: i.firstName + " " + i.lastName, value: i.id }))
            )
            .catch((error) => {
                history.push("/error");
            })
    )

    const onChange = (selected) => {
        setSelected(selected);
        if (null !== selected) {
            history.push(`/patient/${selected.value}`);
            history.go();
        }
    }

    return (
        <div className="container">
            <AsyncSelect
                data-testid="patientSearchBar"
                className="search-bar"
                components={animatedComponent}
                value={selected}
                onChange={onChange}
                isClearable={true}
                placeholder="Search Patients..."
                loadOptions={loadPatients}
                styles={customStyles}
            />
        </div>
    )
}
