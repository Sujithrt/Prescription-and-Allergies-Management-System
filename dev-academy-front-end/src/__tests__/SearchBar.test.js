import React from 'react';
import { shallowToJson } from "enzyme-to-json";
import { shallow, mount } from 'enzyme';
import { Router } from 'react-router-dom';
import SearchBar from '../components/searchBar/SearchBar'

describe("SearchBar Test", () => {
    it("SearchBar is rendered as expected", () => {
        const searchBar = shallow(<SearchBar />);
        expect(shallowToJson(searchBar)).toMatchSnapshot();
    });
});