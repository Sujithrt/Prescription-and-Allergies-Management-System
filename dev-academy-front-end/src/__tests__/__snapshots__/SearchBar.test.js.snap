// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Test SearchBar is rendered as expected 1`] = `
<div
  className="container"
>
  <ForwardRef
    className="search-bar"
    components={
      Object {
        "ClearIndicator": [Function],
        "Control": [Function],
        "CrossIcon": [Function],
        "DownChevron": [Function],
        "DropdownIndicator": [Function],
        "Group": [Function],
        "GroupHeading": [Function],
        "IndicatorSeparator": [Function],
        "IndicatorsContainer": [Function],
        "Input": [Function],
        "LoadingIndicator": [Function],
        "LoadingMessage": [Function],
        "Menu": [Function],
        "MenuList": [Function],
        "MenuPortal": [Function],
        "MultiValue": [Function],
        "MultiValueContainer": [Function],
        "MultiValueLabel": [Function],
        "MultiValueRemove": [Function],
        "NoOptionsMessage": [Function],
        "Option": [Function],
        "Placeholder": [Function],
        "SelectContainer": [Function],
        "SingleValue": [Function],
        "ValueContainer": [Function],
      }
    }
    data-testid="patientSearchBar"
    isClearable={true}
    loadOptions={[Function]}
    onChange={[Function]}
    placeholder="Search Patients..."
    styles={
      Object {
        "control": [Function],
        "option": [Function],
        "singleValue": [Function],
      }
    }
    value=""
  />
</div>
`;
